RevenueCatUI/CustomerCenter/Data/LoadPromotionalOfferUseCase.swift

```diff
@@ -84,22 +84,58 @@ private extension LoadPromotionalOfferUseCase {
     }
 
     private func findDiscount(
```

Comment:
nit: What about renaming this function to `firstEligibleDiscount` and documenting the logic with a test?

Reply:
To do that I would either create a extension on `PromotionalOffer` and inject `purchasesProvider`, or do the oposite and create yet another function in `CustomerCenterPurchasesType`

Reply:
It's being a bit complex to write those tests, but I am on it...